quickwit-cluster/src/cluster.rs

```diff
@@ -193,10 +204,12 @@ impl Cluster {
         self.members.borrow().clone()
     }
 
-    /// Return the grpc addresses corresponding to the list of members.
-    pub async fn members_grpc_addresses(
+    /// Returns the gRPC addresses of the members providing the specified service.
+    /// Returns the addresses of all the members if no service is specified.
+    pub async fn members_grpc_addresses_by_service(
         &self,
         members: &[Member],
```

Comment:
I am not fond of the prototype of this API.

What is the point of the `members` arg? Do we call it for something else than the member of the cluster?

If possible I would rather have:

I'd rather have 
```
pub async fn members_grpc_addresses() -> Vec<SocketAddr>;
pub async fn members_grpc_addresses_for_service(service: QuickwitService) -> Vec<SocketAddr>;
```

Reply:
Unless we change this `members_grpc_addresses(&self) -> Vec<SocketAddr>;` to 
`members_grpc_addresses(&self) -> Vec<(SocketAddr, Vec<QuickwitService>)>; reusing it will be weird as we still need members to get the services, there is also locks involved this is why I preferred a single API